### ISSUE TITLE
Ability to provide config for i18next-express-middleware

### ISSUE THREAD

**Issue by thebat93** [isaachinman/next-i18next#68]
Hi! Thanks for the great library!

I'm experiencing the following problem:
The first request that I send to the Express server sets the language cookie correctly.

![image](https://user-images.githubusercontent.com/29524634/50374478-12ff5200-0600-11e9-88ca-a6491535e9f4.png)

But other requests to internal URLs that include "_next" in their route set the incorrect cookie afterwards.

![image](https://user-images.githubusercontent.com/29524634/50374480-1bf02380-0600-11e9-84f4-2e2ae45d19bf.png)

So, as a result the cookie of the page is the incorrect one.

![image](https://user-images.githubusercontent.com/29524634/50374499-4a6dfe80-0600-11e9-905d-2033a5b1335c.png)

I think I'm experiencing this because each of my requests is processed by the i18next-express-middleware. I noticed that there is a way to specify routes for the middleware that you want to ignore in the config (https://github.com/i18next/i18next-express-middleware#wire-up-i18next-to-request-object) but I didn't find the same functionality here.

Could I suggest adding this feature to the library or maybe there is another way of resolving my issue?

**Comment by isaachinman** [comment]
Very strange, I remember writing [this regex](https://github.com/isaachinman/next-i18next/blob/master/src/middlewares/next-i18next-middleware.js#L14) to fix that exact issue.

Moreover, I cannot seem to reproduce this bug in the [example](https://github.com/isaachinman/next-i18next/tree/master/example):

![screen shot 2018-12-22 at 18 55 53](https://user-images.githubusercontent.com/10575782/50377778-561aee80-061b-11e9-8966-238c42451cd5.png)

Can you please provide a demo repo that reproduce this bug? I'm sure we can get this fixed very quickly. Thanks for your patience.

**Comment by thebat93** [comment]
Thank you for your answer. Here's the minimal setup that I'm using: https://github.com/thebat93/next-i18next-example

I think the problem is that I'm not actually using localePaths. For my project I need to have a locale subpath for the default language too with redirections from paths without locale subpaths in them. When I set localePaths to true the regex that you mentioned is applied and the cookie is set correctly, but I don't get the feature that I want. I saw the open PR with just the feature that I want. Maybe it will be a solution for me.

**Comment by isaachinman** [comment]
@thebat93 Thanks - quick fix with 54f26f2.

In the future we might want to use a `req, res, next` middleware to validate the `ignoreRoutes` setting on `i18next-express-middleware` as this will still be a problem for people using custom routes for assets, etc.

I think to do this we'd need the full list of page routes. As per [this comment](https://github.com/zeit/next.js/issues/3351#issuecomment-348162538), I think we'd have to do something like:

```jsx
const pages = await glob('pages/**/*.js', { cwd: __dirname })
```

Which is not an approach I care to get into at the moment. If more people run into this issue we can readdress it.

**Comment by thebat93** [comment]
Thanks for the quick fix. Unfortunately it resolved the problem only with routes that have '/_next' and '/static' in them, but I also have routes that start with '/fonts' and '/v0'. These routes still mess up the cookie.

**Comment by thebat93** [comment]
I did, but I don't see how it applies to the problem that I need to solve now. The cookie is still set incorrectly. If I use i18next-express-middleware and add ignoreRoutes for '/_next', '/static' and '/fonts' everything works as intended. I just wish there was a way to configure the middleware in your library. Right now I can't fully apply it to my case and have to use the middleware for Express.

**Comment by anthonywebb** [comment]
@thebat93 I am very curious where you added ignoreRoutes?  I'd like to ensure that '/_next', '/static' and '/fonts' are not being processed for my app as well!

**Comment by isaachinman** [comment]
@anthonywebb There's currently no way to do this in `next-i18next` - I assume that @thebat93 meant they were rolling their own implementation.

I can add an options object to the `nextI18NextMiddleware` function, and allow users to pass in routes to ignore. I suppose that's the only sensible thing to do for now. Besides `/static` and `/_next`, it's not safe to ignore any route by default.

**Comment by thebat93** [comment]
It's not possible to provide the config object with ignoreRoutes to the middleware using this library.
But you can simply process all the routes that you need to ignore before passing them to the nextI18NextMiddleware.

So, something like this:
```javascript
const app = next({ dev: process.env.NODE_ENV !== 'production' })
const handle = app.getRequestHandler();

(async () => {
  await app.prepare()
  const server = express()

  // Ignored routes
  server.use(express.static(__dirname + '/public'));

  server.use('/locales', express.static(__dirname + '/locales'));

  server.use('/v0/', someApiMiddleware);

  server.use('/fonts', express.static('./node_modules/fonts/.../'));
  
  // Middleware
  nextI18NextMiddleware(nextI18next, app, server)

  server.get('*', (req, res) => handle(req, res))

  await server.listen(3000)
  console.log('> Ready on http://localhost:3000')
})()
```

**Comment by isaachinman** [comment]
I'll be releasing a fix momentarily.

**Comment by anthonywebb** [comment]
Thanks @thebat93 @isaachinman this will be great.  The interim fix of just including the middleware after the routes that you want to ignore is fine for me so no rush.

**Comment by isaachinman** [comment]
309ed4b

**Comment by isaachinman** [comment]
Released in v0.15.0.
